src/features/QuestionEngine.js

```diff
@@ -0,0 +1,164 @@
+    cqa.createNextAnswer(cqa.question.amountEarlier, cqa.question.amountLater);
+  }
+
+  setCurrentQuestionShown(state, action) {
+    this.currentQuestionAndAnswer(state).latestAnswer.shownTimestamp =
+      action.payload;
+  }
+
+  isLastQandA(state) {
+    return (
+      state.questions.currentQuestionIdx === state.questions.QandA.length - 1
+    );
+  }
+
+  incNextQuestion(state) {
+    if (this.isLastQandA(state)) {
+      state.questions.status = StatusType.Complete;
+    } else {
+      state.questions.currentQuestionIdx += 1;
+      const cqa = this.currentQuestionAndAnswer(state);
+      cqa.createNextAnswer(
+        cqa.question.amountEarlier,
+        cqa.question.amountLater
+      );
+    }
+  }
+
+  updateHighupOrLowdown(currentQandA) {
+    console.assert(
+      currentQandA.question.titration !== TitrationType.none,
+      "Question titration value not set before calling updateHighupOrLowdown."
+    );
+    console.assert(
+      currentQandA.choice !== null &&
+        currentQandA.choice !== ChoiceType.unitialized
+    );
+    const ca = currentQandA.latestAnswer;
+    switch (ca.choice) {
+      case ChoiceType.earlier:
+        var possibleHighup =
+          currentQandA.question.titration === TitrationType.laterAmount
+            ? ca.amountLater
+            : ca.amountEarlier;
+        if (!currentQandA.highup || possibleHighup > currentQandA.highup)
+          currentQandA.highup = possibleHighup;
+        break;
+      case ChoiceType.later:
+        var possibleLowdown =
+          currentQandA.question.titration === TitrationType.laterAmount
+            ? ca.amountLater
+            : ca.amountEarlier;
+        if (!currentQandA.lowdown || possibleLowdown < currentQandA.lowdown)
+          currentQandA.lowdown = possibleLowdown;
+        break;
+      default:
+        console.assert(
+          true,
+          "Invalid value for current answer in setAnswerCurrentQuestion"
+        );
+        break;
+    }
+  }
+
+  calcTitrationAmount(titratingAmount, highup, override) {
+    return (titratingAmount ? titratingAmount - highup : override) / 2;
+  }
+
+  calcNewAmount(QandA, titrationAmount) {
+    var adjustmentAmount;
+    switch (QandA.question.titration) {
+      case TitrationType.laterAmount:
+        console.assert(
+          QandA.latestAnswer.choice &&
+            QandA.latestAnswer.choice !== ChoiceType.unitialized
+        );
+        adjustmentAmount =
+          QandA.latestAnswer.choice === ChoiceType.earlier
+            ? titrationAmount
+            : -1 * titrationAmount;
+        return (
+          parseInt((QandA.latestAnswer.amountLater + adjustmentAmount) / 10) *
+          10
+        );
+      case TitrationType.earlierAmount:
+        adjustmentAmount =
+          QandA.latestAnswer.choice === ChoiceType.earlier
+            ? -1 * titrationAmount
+            : titrationAmount;
+        return (
+          parseInt((QandA.latestAnswer.amountEarlier + adjustmentAmount) / 10) *
```

Comment:
reusing same formula from
https://github.com/pcordone/vizsurvey/pull/10/files#diff-6a998fe333db41abea4c9933561cf59fc1e3b6d8d3b41cd23b3da1a861fe2275R107
and not sure if there is a way to make it a bit more readable by pulling this out of the switch case statements and simplifying some of the assignments 🤷

Reply:
Yea, not sure how to make this simpler.  Let me think about it.